media-sound/audacity/audacity-3.2.2.ebuild

```diff
@@ -0,0 +1,234 @@
+#   in our best interest to fix that as a patch series.
+# glib, gtk and gdk are all directly relied on in the source, not just
+#   through wxGTK
+
+# These are needed for Sentry and audio.com.
+MY_NETDEPEND="
+	net-misc/curl
+	dev-libs/rapidjson
+"
+
+RDEPEND="dev-db/sqlite:3
+	dev-libs/expat
+	dev-libs/glib:2
+	media-libs/libsndfile
+	media-libs/libsoundtouch:=
+	media-libs/portaudio[alsa?]
+	media-libs/portmidi
+	media-libs/portsmf:=
+	media-libs/soxr
+	>=media-sound/lame-3.100-r3
+	x11-libs/gdk-pixbuf:2
+	x11-libs/gtk+:3
+	x11-libs/wxGTK:${WX_GTK_VER}[X]
+	sys-apps/util-linux
+	sys-libs/zlib
+	alsa? ( media-libs/alsa-lib )
+	ffmpeg? ( media-video/ffmpeg )
+	flac? ( media-libs/flac:=[cxx] )
+	id3tag? ( media-libs/libid3tag:= )
+	lv2? (
+		dev-libs/serd
+		dev-libs/sord
+		>=media-libs/lilv-0.24.6-r2
+		media-libs/lv2
+		media-libs/sratom
+		media-libs/suil
+	)
+	mad? ( >=media-libs/libmad-0.15.1b )
+	mpg123? ( media-sound/mpg123 )
+	ogg? ( media-libs/libogg )
+	opus? ( media-libs/opus )
+	sbsms? ( media-libs/libsbsms )
+	telemetry? ( ${MY_NETDEPEND} )
+	audiocom? ( ${MY_NETDEPEND} )
+	twolame? ( media-sound/twolame )
+	vamp? ( media-libs/vamp-plugin-sdk )
+	vorbis? ( media-libs/libvorbis )
+	wavpack? ( media-sound/wavpack )
+"
+DEPEND="${RDEPEND}"
+BDEPEND="app-arch/unzip
+	sys-devel/gettext
+	virtual/pkgconfig
+"
+
+REQUIRED_USE="
+	mad? ( !mpg123 )
+"
+
+PATCHES=(
+	# Equivalent to previous versions
+	"${FILESDIR}/${PN}-3.2.2-disable-ccache.patch"
+	# From Debian
+	"${FILESDIR}/${PN}-3.2.2-fix-rpaths.patch"
+
+	# Disables some header-based detection
+	"${FILESDIR}/${PN}-3.2.2-allow-overriding-alsa-jack.patch"
+
+	# Allows re-enabling telemetry
+	"${FILESDIR}/${PN}-3.2.2-remove-conan-threadpool.patch"
+	"${FILESDIR}/${PN}-3.2.2-allow-finding-rapidjson-pkgconfig.patch"
+)
+
+src_prepare() {
+	cmake_src_prepare
+
+	local header_subs="${S}/lib-src/header-substitutes"
+	cat <<-EOF >"${header_subs}/allegro.h" || die
+/* Hack the allegro.h header substitute to use system headers.  */
+#include <portsmf/allegro.h>
+EOF
+
+	# Keep condition in sync with has_networking in src_configure.
+	if use telemetry || use audiocom; then
+		mkdir -p "${S}/"/lib-src/threadpool/ThreadPool/ || die
+		cp "${DISTDIR}"/progschj-ThreadPool-"${MY_THREADPOOL_DATE}".h \
+		   "${S}"/lib-src/threadpool/ThreadPool/ThreadPool.h || die
+	fi
+}
+
+# NOTES:
+# I traced a bunch of dlopens via LD_AUDIT, and concluded that
+# dev-libs/{serd,sord}, media-libs/{opus,sratom}, media-sound/lame,
+# media-video/ffmpeg (yes), sys-libs/zlib are all used via dlopen.
+# FFmpeg in particular could even lose it's slot dependency, because
+# Audacity picks which version of the ABI it interacts with at runtime,
+# but I'm not sure how good of an idea that is.  Anyway - for this
+# release, the supported Lavc versions are 5[789], Lavu are 5[2567], and
+# Lavf are 5[789].  Check libraries/lib-ffmpeg-support/impl if you want
+# to check.  It's also useful to grep for wxDynamicLibrary and dlsym.
```

Comment:
I kind of find this out of place, I believe it'd better be suited close to the `*DEPEND` blocks. I'm also not a huge fan of the "blog" text in the beginning, but at least it does make a point, if you add some of the deps as unconditional ones or leave the subslot binder out from ffmpeg.

Reply:
Reworded this whole thing and organized it better.  This rambling style block was pretty much precisely that - I was in the process of getting a hang of what it's doing with dlopens when writing this.  Thanks for nudging me to organize it.